### ISSUE TITLE
Controller tests dont run after adding auth middleware

### ISSUE THREAD

**Issue by Cnordg** [sopra-fs23-group-13/meme-it-server#102]
Error:
`java.lang.IllegalStateException: Failed to load ApplicationContext`

Details:
```Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jwtRequestFilter' defined in file [/Users/henriknordgren/Documents/Documents - Local-P213124X/UZH/Semester 4 (FS2023)/Software Engineering Lab/meme-it-server/bin/main/ch/uzh/ifi/hase/soprafs23/JwtRequestFilter.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'ch.uzh.ifi.hase.soprafs23.repository.UserRepository' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Qualifier("userRepository")}```

**Comment by Cnordg** [comment]
We need to disable the spring boot security when running the controller, but somehow none of the [solutions](https://stackoverflow.com/questions/31169720/disable-security-for-unit-tests-with-spring-boot) work

**Comment by Cnordg** [comment]
More info regarding error [Spring failed to load applicationcontext](https://www.baeldung.com/spring-junit-failed-to-load-applicationcontext)

**Comment by Cnordg** [comment]
This stack overflow [question](https://stackoverflow.com/questions/31169720/disable-security-for-unit-tests-with-spring-boot) might give more info too

**Comment by weberlii** [comment]
This is the most promising approach so far:
https://stackoverflow.com/questions/55522098/no-qualifying-bean-of-type-mypackage-repository-available-expected-at-least-1

**Comment by weberlii** [comment]
The primary issue here was that when the org.springframework.boot:spring-boot-starter-security was added, this automatically adds specific "security" beans into the Spring ApplicationContext, therefore the MVCTests will expect you to pass the authorization checks as what was being seen.
The first step here is to disable the "security configuration" by adding the following lines into those MvcTests:

excludeAutoConfiguration = {
                SecurityAutoConfiguration.class
        }
The next bit of complexity came about as there is a specific JwtSecurityConfig configuration bean, as this involves more security configuration, we needed to disable this too, hence the

@WebMvcTest(controllers = LobbyController.class,
        excludeAutoConfiguration = {
                SecurityAutoConfiguration.class,
                JwtSecurityConfig.class
        })
In addition to this, I noticed the JwtRequestFilter was being registered as a @Component outside of the JwtSecurityConfig, this meant the "security" of the application was configured in two places, I have now encapsulated the wiring up of the JwtRequestFilter as Bean inside the JwtSecurityConfig class. This allows us to now exclude this auto configuration as per the lines above.

Lastly, some assertions were incorrect as the value asserting was not what was expected.
